### PR TITLE
REL: 1.2.2

### PR DESCRIPTION
**Pull request by oesteban**
## Summary

Prep for 1.2.2 release, targeting Friday, September 6.

Please add any issues you want to see included in the following release to the [upcoming 1.2.3 milestone](https://github.com/nipy/nipype/milestone/35) targetting September 23.

## Release checklist

* [x] Update changelog
* [x] Update .mailmap
* [x] Update .zenodo.json
* [x] Set release number in `nipype/info.py` and `doc/conf.py`
* [x] Check `doc/documentation.rst` with previous releases
* [x] Check conda-forge feedstock build (conda-forge/nipype-feedstock#48)
* [ ] Tutorial tests pass - https://circleci.com/workflow-run/37f1cebd-69f8-40eb-80a5-76e001e55b3b

## Uncredited authors

The following authors have contributed, but not added themselves to the [`.zenodo.json`](https://github.com/nipy/nipype/blob/master/.zenodo.json) file. If you would like to be an author on Zenodo releases, please add yourself or comment with your preferred publication name, affiliation and [ORCID](https://orcid.org/). If you would like to stop being spammed whenever I'm the one doing releases, let me know, and I'll add you to a blacklist.

No entry to sort: Victor Férat (@vferat)
~~No entry to sort: Murat Bilgel (@bilgelm)~~

## Acknowledgment

- [x] \(Mandatory\) I acknowledge that this contribution will be available under the Apache 2 license.

### CODE REVIEW

File: doc/changelog/1.X.X-changelog [hunk 1]
```diff
@@ -1,3 +1,13 @@
+1.2.2 (September 05, 2019)
+==========================
+
```
Comment (effigies) [inline doc/changelog/1.X.X-changelog]: We've been linking the milestone (see L14).

**Comment by effigies** [comment]
@djarecka Oscar didn't ping you, but if you get a chance to verify that we're not breaking the tutorials, could you do that? Or show one of us how you do that?

**Comment by satra** [comment]
@djarecka  and @effigies -  we should add a test to nipype that clones the tutorial repo and runs the tests automatically using `nbconvert`, we are doing a version of this in pydra. i just don't know how long that test will run, so perhaps it can be a test that's only run prior to release.

**Comment by oesteban** [comment]
@djarecka just in case coming back from INCF allows you, please let us know how we can check we are not breaking the tutorials.

@effigies not sure what is next w.r.t. conda-forge

@satra @mgxd et al. thumbs up?

**Comment by effigies** [comment]
Nothing to do at conda forge except check that it passes. Once you release, it'll update with the new hash and we can merge.

Also, don't know if I mentioned it, but you should push an annotated tag before making the GitHub release.

**Comment by effigies** [comment]
@oesteban I've started a job here: https://circleci.com/workflow-run/37f1cebd-69f8-40eb-80a5-76e001e55b3b. It should run on the latest master, so if it passes, we're set.

**Comment by oesteban** [comment]
Should I push the annotated tag before merging this PR?

**Comment by effigies** [comment]
No. Merge, then:

```Shell
git fetch upstream
git checkout upstream/master
git tag -a 1.2.2 # -s if you want to sign with a GPG key
git push upstream 1.2.2
```

For the tag message, I just copy the changelog, minus any Markdown stuff.

e.g.

```
$ git show 1.2.1 | cat
tag 1.2.1
Tagger: Christopher J. Markiewicz <markiewicz@stanford.edu>
Date:   Mon Aug 19 08:35:50 2019 -0400

1.2.1 (August 19, 2019)

Full changelog: https://github.com/nipy/nipype/milestone/32?closed=1

  * FIX: Resolve/rebase paths from/to results files (https://github.com/nipy/nipype/pull/2971)
  * FIX: Use ``load_resultfile`` when loading a results pickle (https://github.com/nipy/nipype/pull/2985)
  * FIX: Incorrect extension identified when checking ``File`` traits (https://github.com/nipy/nipype/pull/2987)
  * FIX: Correctly pickle ``OuputMultiObject`` traits (https://github.com/nipy/nipype/pull/2983)
  * FIX: Improve output handling in DWIDenoise and DWIBiasCorrect (https://github.com/nipy/nipype/pull/2978)
  * FIX: Docker build (https://github.com/nipy/nipype/pull/2963)
  * FIX: Remove '=' signs from EddyQuad argument specifications (https://github.com/nipy/nipype/pull/2941)
  * FIX: Set input model to bedpostx for camino.TrackBedpostxProba (https://github.com/nipy/nipype/pull/2947)
  * FIX: Allow ``max_sh``not to be set (auto mode) (https://github.com/nipy/nipype/pull/2940)
  * ENH: Update mrtrix reconst.py EstimateFOD max_sh to be able to accept list (https://github.com/nipy/nipype/pull/2990)
  * ENH: Let ``indirectory`` handle ``nipype.utils.filemanip.Path`` (https://github.com/nipy/nipype/pull/2989)
  * ENH: Add resolve/rebase ``BasePath`` traits methods & tests (https://github.com/nipy/nipype/pull/2970)
  * ENH: Modify ``Directory`` and ``File`` traits to get along with pathlib (https://github.com/nipy/nipype/pull/2962)
  * REF: Update nipype2boutiques script (https://github.com/nipy/nipype/pull/2894)
  * TST: Parametrize JoinNode expansion tests over config ``needed_outputs`` (https://github.com/nipy/nipype/pull/2981)
  * MAINT: Pin lxml<4.4.0 for Python 3.4 (https://github.com/nipy/nipype/pull/2980)
  * MAINT: Refactor ``aggregate_outputs`` for readability (https://github.com/nipy/nipype/pull/2969)
  * MAINT: Bump neurodocker version (https://github.com/nipy/nipype/pull/2965)
  * MAINT: Various minor improvements to complement previous PR (https://github.com/nipy/nipype/pull/2964)
  * MAINT: Sort dependencies alphabetically (https://github.com/nipy/nipype/pull/2961)
-----BEGIN PGP SIGNATURE-----

iI4EABYIADYWIQR8q78A2LscOhhE3OS5m8FQiGeN7gUCXVqXvxgcbWFya2lld2lj
ekBzdGFuZm9yZC5lZHUACgkQuZvBUIhnje5ycQEAw/nJruct3f6baCgginX2tzq4
SqcoPMj7IXi2dJvG0sMBAPrzraRmJQh+SRkugRrU0f8FsLhgsEFH7LKIKth5JscL
=jjn7
-----END PGP SIGNATURE-----

commit 3e794d414c0bd79d077fc5ba36f2112e7b95878b
Merge: ea2243cef 17019b417
Author: Chris Markiewicz <markiewicz@stanford.edu>
Date:   Mon Aug 19 05:33:58 2019 -0700

    Merge pull request #2999 from nipy/rel/1.2.1
    
    REL: 1.2.1

```

**Comment by effigies** [comment]
If you mess up the tag, you can always just `git tag --delete 1.2.2`, so feel free to play around until you get it how you want.

**Comment by oesteban** [comment]
Bummer, I missed #2948 and #3006. I'll fix that before releasing (so I'll need to delete the tag, etc.)

**Comment by effigies** [comment]
I cancelled the Circle build. You can delete the tag on the repo with:

```
git push upstream :1.2.2
```

File: doc/changelog/1.X.X-changelog [hunk 2]
```diff
@@ -1,3 +1,17 @@
+1.2.2 (September 05, 2019)
```
Comment (effigies) [inline doc/changelog/1.X.X-changelog]: @oesteban If you're pushing a new tag, you might want to update the date before tagging.

**Pull request merged**